independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/AbstractGradleBuildFile.java

```diff
@@ -61,11 +61,30 @@ protected boolean addDependency(AppArtifactCoords coords, boolean managed) {
     }
 
     static boolean addDependencyInModel(Model model, AppArtifactCoords coords, boolean managed) {
-        StringBuilder newDependency = new StringBuilder()
-                .append("    implementation '")
-                .append(coords.getGroupId())
-                .append(":")
-                .append(coords.getArtifactId());
+        boolean isBOM = "pom".equals(coords.getType());
+        StringBuilder newDependency;
```

Comment:
It seems really weird to have the quarkus bom insertion as part of the `addDependencyInModel `

Reply:
Why? You can have other BOMs when using the extension registry

Reply:
aww right, I forgot about that :)

Reply:
The extension registry creates an `ExtensionInstallPlan` with the extensions and BOMs needed to work, which then it's up to the `ExtensionManager` to decide what needs to be added/changed/etc

Reply:
@gastaldi do you use the currently installed extensions to create that plan?

Reply:
No, it's up to the `ExtensionManager` to decide what to do with the `ExtensionInstallPlan` and the currently installed extensions

Reply:
ok ;-) I just wanted to make sure that my other comment about the version was relevant..